From 2aaa9bb1bfa76bea1228629b8874051eec16a33b Mon Sep 17 00:00:00 2001
From: Flavio Martins <flavioxmartins@gmail.com>
Date: Mon, 6 May 2024 13:24:54 +0100
Subject: [PATCH] Disable jemalloc on linux

---
 Cargo.toml  | 2 +-
 src/main.rs | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/Cargo.toml b/Cargo.toml
index c15ba52..d976397 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -73,7 +73,7 @@ libc = "0.2"
 # FIXME: Re-enable jemalloc on macOS
 # jemalloc is currently disabled on macOS due to a bug in jemalloc in combination with macOS
 # Catalina. See https://github.com/sharkdp/fd/issues/498 for details.
-[target.'cfg(all(not(windows), not(target_os = "android"), not(target_os = "macos"), not(target_os = "freebsd"), not(target_os = "openbsd"), not(all(target_env = "musl", target_pointer_width = "32")), not(target_arch = "riscv64")))'.dependencies]
+[target.'cfg(all(not(windows), not(target_os = "android"), not(target_os = "macos"), not(target_os = "freebsd"), not(target_os = "openbsd"), not(all(target_env = "musl", target_pointer_width = "32")), not(target_arch = "riscv64"), not(target_os = "linux")))'.dependencies]
 jemallocator = {version = "0.5.4", optional = true}

 [dev-dependencies]
diff --git a/src/main.rs b/src/main.rs
index bef4120..d80317b 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -43,6 +43,7 @@ use crate::regex_helper::{pattern_has_uppercase_char, pattern_matches_strings_wi
     not(target_os = "openbsd"),
     not(all(target_env = "musl", target_pointer_width = "32")),
     not(target_arch = "riscv64"),
+    not(target_os = "linux"),
     feature = "use-jemalloc"
 ))]
 #[global_allocator]
--
2.45.0

