From 3a402f1a51e6e68e6191e68a47e0a457bf798d1a Mon Sep 17 00:00:00 2001
From: Daniel Bast <2790401+dbast@users.noreply.github.com>
Date: Mon, 8 Sep 2025 14:57:34 +0200
Subject: [PATCH] [PATCH] Disable jemalloc on linux

---
 Cargo.toml  | 2 +-
 src/main.rs | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/Cargo.toml b/Cargo.toml
index 077f072..c732fe8 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -74,7 +74,7 @@ libc = "0.2"
 # Catalina. See https://github.com/sharkdp/fd/issues/498 for details.
 # This has to be kept in sync with src/main.rs where the allocator for
 # the program is set.
-[target.'cfg(all(not(windows), not(target_os = "android"), not(target_os = "macos"), not(target_os = "freebsd"), not(target_os = "openbsd"), not(target_os = "illumos"), not(all(target_env = "musl", target_pointer_width = "32")), not(target_arch = "riscv64")))'.dependencies]
+[target.'cfg(all(not(windows), not(target_os = "android"), not(target_os = "macos"), not(target_os = "freebsd"), not(target_os = "openbsd"), not(target_os = "illumos"), not(all(target_env = "musl", target_pointer_width = "32")), not(target_arch = "riscv64"), not(target_os = "linux")))'.dependencies]
 tikv-jemallocator = {version = "0.6.0", optional = true}
 
 [dev-dependencies]
diff --git a/src/main.rs b/src/main.rs
index 94623eb..87a4fb8 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -47,6 +47,7 @@ use crate::regex_helper::{pattern_has_uppercase_char, pattern_matches_strings_wi
     not(target_os = "illumos"),
     not(all(target_env = "musl", target_pointer_width = "32")),
     not(target_arch = "riscv64"),
+    not(target_os = "linux"),
     feature = "use-jemalloc"
 ))]
 #[global_allocator]
-- 
2.51.0

